frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -1,14 +1,59 @@
-
 import * as _ from 'lodash-es';
 import * as React from 'react';
 import * as fuzzy from 'fuzzysearch';
-
+import { getName } from '@console/shared';
 import { Firehose, LoadingInline, Dropdown, ResourceName, ResourceIcon } from '.';
 import { isDefaultClass } from '../storage-class';
+import { StorageClass } from '../storage-class-form';
 
-/* Component StorageClassDropdown - creates a dropdown list of storage classes */
+type StorageClassFilter = (resource: StorageClass) => boolean;
+
+type StorageClassDropdownItems = {
+  [key: string]: {
+    kindLabel: string;
+    name: string;
+    description?: string;
+    default?: boolean;
+    accessMode?: string;
+    provisioner?: string;
+    type?: string;
+    zone?: string;
+  };
+};
 
-export class StorageClassDropdownInner extends React.Component<StorageClassDropdownInnerProps, StorageClassDropdownInnerState> {
+const getUnorderedItems = (
+  items: StorageClass[],
+  filter: StorageClassFilter = (resource) => _.isEmpty(resource),
+): StorageClassDropdownItems =>
+  items.reduce(
+    (acc, resource) => ({
+      ...acc,
+      ...(filter(resource) && {
+        [getName(resource)]: {
+          kindLabel: 'StorageClass',
+          name: getName(resource),
+          description: _.get(resource, 'metadata.annotations.description', ''),
```

Comment:
You do want explicit `''` empty string as default?

Reply:
I didn't make any explicit changes here. It was using empty string by default before I touched this code.